ui/src/components/admin/AdminWhitelistManager.js

```diff
@@ -0,0 +1,92 @@
+import {
+  Heading,
+  Text,
+  Container,
+  Textarea,
+  Button,
+  useToast,
+} from '@chakra-ui/react';
+import { useForm } from 'react-hook-form';
+import { customAxios as axios } from '../helpers/customAxios';
+
+const AdminWhitelistManager = (props) => {
+  const { register, handleSubmit, reset } = useForm();
+  const toast = useToast();
+
+  const submitWhitelist = async (formData) => {
+    try {
+      let arrayOfEmails = formData.whitelistData.split('\n');
+      let { data } = await axios.get('/api/whitelist/get');
+      if (data.length === 0) {
+        let { status } = await axios.post('/api/whitelist/create', {
+          eventName: 'ignite-2022',
+          data: arrayOfEmails,
+        });
+        if (status === 200) {
+          toast({
+            title: 'Success!',
+            description: 'Whitelist created.',
+            status: 'success',
+            duration: 3000,
+            isClosable: true,
+          });
+        }
+        reset();
+      } else {
+        let igniteEntry = data.filter((obj) => obj.eventName === 'ignite-2022');
+        if (igniteEntry.length === 1) {
+          let { status } = await axios.post('/api/whitelist/update', {
```

Comment:
when testing, updating the whitelist overwrites the past records. not sure if this is intended behaviour

Reply:
yeah that is the intended behaviour